OSIRIS/reduce.py

```diff
@@ -1 +1,86 @@
-raise Exception('script not implemented')
+# import mantid algorithms, numpy and matplotlib
+from mantid.simpleapi import *
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def generate_input_path_for_run(run_number, cycle):
+    return f"/archive/ndxosiris/Instrument/data/{cycle}/OSI{run_number}.nxs"
+
+# To change by automatic script
+input_runs = ["108538", "108539"]
+# This needs to be loaded from a shared repository of files
+calibration_file_path = "/extras/osi92682_multi_graphite002_calib.nxs"
+cycle = "cycle_14_1"
+analyser = "graphite"
+reflection = "002"
+spectroscopy_reduction = True
+diffraction_reduction = True
+
+# Defaults
+instrument = "OSIRIS"
+workspace_start_of_file = ""
+efixed = 1.845
+spec_spectra_range = "963,1004"
+diff_spectra_range = '3,962'
+# Grouping string is for 14 groups, and generated by the UI for indirect data reduction and based on spectra_range
+grouping_string = "963-965,966-968,969-971,972-974,975-977,978-980,981-983,984-986,987-989,990-992,993-995,996-998,999-1001,1002-1004"
+unit_x = "DeltaE"
+fold_multiple_frames = False
+
+# Generated
+sum_runs = len(input_runs) > 1
+
+input_file_paths = ""
+for input_run in input_runs:
+    input_file_paths += ", " + generate_input_path_for_run(input_run, cycle)
+input_file_paths = input_file_paths[2:]
+print(input_file_paths)
+
+output_workspaces = []
+
+# Perform the reduction
+if spectroscopy_reduction:
+    output_workspace_prefix = instrument
+    for input_run in input_runs:
+        output_workspace_prefix += input_run + ", "
+    output_workspace_prefix = output_workspace_prefix[:-2] + f"_{analyser}_{reflection}_Reduced"
```

Comment:
Similar to above the final 2 char index feels magic number-ish

Reply:
```suggestion
    output_workspace_prefix = output_workspace_prefix[:-2] + f"_{analyser}_{reflection}_Reduced"  # Slice out the excess ", " and finalize prefix
```